Sensor: mpr: Add Honeywell MPR driver and example

sven-hm opened the pull request:
The Honeywell MPR sensor is a piezoresistive silicon pressure sensor,
which can be accessed via i2c.

https://sensing.honeywell.com/micropressure-mpr-series

Signed-off-by: Sven Herrmann <sven.herrmann@posteo.de>

galak commented on drivers/sensor/mpr/mpr.c:
```diff
@@ -0,0 +1,139 @@
+};
+
+static struct mpr_data mpr_data;
+static const struct mpr_config mpr_cfg = {
```
Please use the new DT_INST macro's in include/devicetree.h (see example in drivers/sensor/fxos8700)

sven-hm replied:
Ok, done.

galak commented on dts/bindings/sensor/honeywell,mpr.yaml:
```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: Apache-2.0
+
+description: Honeywell Digital Pressure Sensor
+
```
Is there a datasheet URL or something you can possible reference here?

sven-hm replied:
Ok, added.

galak reviewed (commented):
You'll need to add a dts node in tests/drivers/build_all/i2c.dtsi

MaureenHelm commented on drivers/sensor/mpr/Kconfig:
```diff
@@ -0,0 +1,124 @@
+config MPR_OUTPUT_TYPE_6
+	bool "i2c, address 0x68"
+config MPR_OUTPUT_TYPE_7
+	bool "i2c, address 0x78"
```
this should come from device tree, not kconfig

sven-hm replied:
Ok, removed.

MaureenHelm commented on drivers/sensor/mpr/mpr.c:
```diff
@@ -0,0 +1,137 @@
+
+LOG_MODULE_REGISTER(MPR, CONFIG_SENSOR_LOG_LEVEL);
+
+int mpr_init(struct device *dev)
```
make this `static`

sven-hm replied:
Ok, done.

MaureenHelm commented on drivers/sensor/mpr/mpr.c:
```diff
@@ -0,0 +1,137 @@
+
+	data->i2c_master = device_get_binding(cfg->i2c_bus);
+	if (!data->i2c_master) {
+		LOG_DBG("mpr: i2c master not found: %s", cfg->i2c_bus);
```
replace `LOG_DBG` with `LOG_ERR`

sven-hm replied:
Ok, done.

MaureenHelm commented on samples/sensor/mpr/CMakeLists.txt:
```diff
@@ -0,0 +1,8 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
```
please move the sample to a separate commit. you can keep it in the same PR though

sven-hm replied:
Ok, done.

MaureenHelm commented on samples/sensor/mpr/sample.yaml:
```diff
@@ -0,0 +1,8 @@
+    harness: sensor
+    tags: sensors
+    depends_on: i2c
+    filter: dt_compat_enabled("honeywell,mpr")
```
we don't have any boards with this compatible, so this sample isn't going to be built in CI. i usually don't like whitelists, but in this case i think you should add arduino_due as a platform_whitelist instead of the filter.

sven-hm replied:
Ok, done.

MaureenHelm reviewed (changes requested):
thanks for the contribution!

please squash the fixup commit into the previous commit, and rebase to the latest master to fix the merge conflict.

sven-hm committed:
Sensor: mpr: Add Honeywell MPR driver

The Honeywell MPR sensor is a piezoresistive silicon pressure sensor,
which can be accessed via i2c.

https://sensing.honeywell.com/micropressure-mpr-series

Signed-off-by: Sven Herrmann <sven.herrmann@posteo.de>

sven-hm committed:
sample: sensor: mpr: Add MPR sample application

The sample reads pressure data from an MPR sensor and prints it to the
console.

Signed-off-by: Sven Herrmann <sven.herrmann@posteo.de>

sven-hm commented:
The PR is now split into two commits, one for the driver the other for the sample. I cleaned up the header and Kconfig files and added a node in tests/drivers/build_all/i2c.dtsi.
